Fix content pack dependency resolving for input entities

Every content pack facade needs to handle the resolving of its own dependencies to make sure the content pack service includes all dependencies when generating a content pack.

The [InputFacade](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java) is currently not doing this for lookup table extractors and converters. A lookup table extractor/converter references a lookup table so the InputFacade needs to make sure to include the lookup table in the dependency graph.

That basically means InputFacade needs to implement [EntityFacade#resolveNativeEntity() and EntityFacade#resolveForInstallation()](https://github.com/Graylog2/graylog2-server/blob/0dbe6f659e79e945662658bc9c2cd468c8922b8c/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/EntityFacade.java#L108-L134) to put the required lookup table into the dependency graph.

The [LookupTableFacade](https://github.com/Graylog2/graylog2-server/blob/master/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/LookupTableFacade.java) is basically doing the same for caches and data adapters and can server as an example on how to do this.

Due to this bug, installing a content pack with an input that has a lookup table converter and/or extractor doesn't work right now and throws the following error:

```
2018-09-13 18:51:07,854 ERROR: org.graylog2.shared.rest.exceptionmappers.AnyExceptionClassMapper - Unhandled exception in REST resource
org.graylog2.contentpacks.exceptions.ContentPackException: Failed to install content pack <a0de9fa3-a860-8cfe-08ec-911f513d07c3/1>
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:147) ~[classes/:?]
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:98) ~[classes/:?]
	at org.graylog2.rest.resources.system.contentpacks.ContentPackResource.installContentPack(ContentPackResource.java:273) ~[classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_181]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) ~[jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) [jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [jersey-common-2.25.1.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [jersey-common-2.25.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [jersey-common-2.25.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [jersey-common-2.25.1.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [jersey-common-2.25.1.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) [jersey-common-2.25.1.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) [jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) [jersey-server-2.25.1.jar:?]
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384) [jersey-container-grizzly2-http-2.25.1.jar:?]
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224) [grizzly-http-server-2.3.28.jar:2.3.28]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:181) [metrics-core-4.0.2.jar:4.0.2]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_181]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
Caused by: java.lang.RuntimeException: Couldn't create extractors
	at org.graylog2.contentpacks.facades.InputFacade.decode(InputFacade.java:223) ~[classes/:?]
	at org.graylog2.contentpacks.facades.InputFacade.createNativeEntity(InputFacade.java:185) ~[classes/:?]
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:138) ~[classes/:?]
	... 29 more
Caused by: java.lang.IllegalStateException: Configured lookup table <dns-a> doesn't exist
	at org.graylog2.inputs.extractors.LookupTableExtractor.<init>(LookupTableExtractor.java:57) ~[classes/:?]
	at org.graylog2.inputs.extractors.ExtractorFactory.factory(ExtractorFactory.java:72) ~[classes/:?]
	at org.graylog2.contentpacks.facades.InputFacade.addExtractor(InputFacade.java:308) ~[classes/:?]
	at org.graylog2.contentpacks.facades.InputFacade.createExtractors(InputFacade.java:273) ~[classes/:?]
	at org.graylog2.contentpacks.facades.InputFacade.decode(InputFacade.java:221) ~[classes/:?]
	at org.graylog2.contentpacks.facades.InputFacade.createNativeEntity(InputFacade.java:185) ~[classes/:?]
	at org.graylog2.contentpacks.ContentPackService.installContentPack(ContentPackService.java:138) ~[classes/:?]
	... 29 more
```

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -26,6 +26,7 @@ import com.google.common.graph.ImmutableGraph;
 import com.google.common.graph.MutableGraph;
 import com.google.common.primitives.Ints;
 import org.graylog2.contentpacks.exceptions.ContentPackException;
+import org.graylog2.contentpacks.jackson.ReferenceConverter;
 import org.graylog2.contentpacks.model.ModelId;
 import org.graylog2.contentpacks.model.ModelType;
 import org.graylog2.contentpacks.model.ModelTypes;
@@ -39,6 +40,7 @@ import org.graylog2.contentpacks.model.entities.EntityV1;
 import org.graylog2.contentpacks.model.entities.EntityWithConstraints;
 import org.graylog2.contentpacks.model.entities.ExtractorEntity;
 import org.graylog2.contentpacks.model.entities.InputEntity;
+import org.graylog2.contentpacks.model.entities.LookupTableEntity;
 import org.graylog2.contentpacks.model.entities.NativeEntity;
 import org.graylog2.contentpacks.model.entities.NativeEntityDescriptor;
 import org.graylog2.contentpacks.model.entities.references.ReferenceMap;
@@ -487,5 +489,40 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
             LOG.debug("Couldn't find input {}", entityDescriptor, e);
         }
         return ImmutableGraph.copyOf(mutableGraph);
+    }
+
+    @Override
+    public Graph<Entity> resolveForInstallation(Entity entity,
+                                                Map<String, ValueReference> parameters,
+                                                Map<EntityDescriptor, Entity> entities) {
+        if(entity instanceof EntityV1) {
+            return resolveForInstallationV1((EntityV1) entity, parameters, entities);
+        } else {
+            throw new IllegalArgumentException("Unsupported entity version: " + entity.getClass());
+        }
+    }
+
+    private Graph<Entity> resolveForInstallationV1(EntityV1 entity,
+                                                   Map<String, ValueReference> parameters,
+                                                   Map<EntityDescriptor, Entity> entities) {
+        final MutableGraph graph = GraphBuilder.directed().build();
+        graph.addNode(entity);
+
+        final InputEntity input = objectMapper.convertValue(entity.data(), InputEntity.class);
+        final Set<String> lookupTableNames = input.extractors().stream()
+                .filter(e -> e.type().asString(parameters).equals(Extractor.Type.LOOKUP_TABLE.toString()))
+                .map(e -> e.configuration())
+                .map(c -> ((ValueReference) c.get(LookupTableExtractor.CONFIG_LUT_NAME)).asString(parameters))
+                .collect(Collectors.toSet());
+        entities.entrySet().stream()
+                .filter(x -> x.getValue().type().equals(ModelTypes.LOOKUP_TABLE_V1))
+                .filter(x -> {
+                    EntityV1 entityV1 = (EntityV1) x.getValue();
+                    LookupTableEntity lookupTableEntity = objectMapper.convertValue(entityV1.data(), LookupTableEntity.class);
+                    return  lookupTableNames.contains(lookupTableEntity.name().asString(parameters));
+                })
+                .forEach(x -> graph.putEdge(entity, x.getValue()));
+
+        return ImmutableGraph.copyOf(graph);
     }
 }

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -528,7 +528,9 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                 .collect(Collectors.toSet());
 
         input.extractors().stream().map(c -> c.converters()
-                .stream().map(con -> ((ValueReference) con.configuration().get("lookup_table_name")).asString(parameters))
+                .stream()
+                .filter(con -> con.type().asString(parameters).equals(Converter.Type.LOOKUP_TABLE.name()))
+                .map(con -> ((ValueReference) con.configuration().get("lookup_table_name")).asString(parameters))
                 .collect(Collectors.toSet()))
                 .forEach(lookupTableNames::addAll);
 

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -485,6 +485,19 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                         });
                     });
 
+            inputWithExtractors.extractors().stream()
+                    .filter(e -> e.getType().equals(Extractor.Type.LOOKUP_TABLE))
+                    .forEach(e -> e.getConverters().stream().filter(c -> c.getType().equals(Converter.Type.LOOKUP_TABLE))
+                            .map(c -> (String) c.getConfig().get("lookup_table_name"))
+                            .map(name -> lookupTableService.get(name))
+                            .forEach(optional -> {
+                                optional.ifPresent(lookupTableDto -> {
+                                    EntityDescriptor lookupTable = EntityDescriptor.create(
+                                            ModelId.of(lookupTableDto.id()), ModelTypes.LOOKUP_TABLE_V1);
+                                    mutableGraph.putEdge(entityDescriptor, lookupTable);
+                                });
+                            }));
+
         } catch (NotFoundException e) {
             LOG.debug("Couldn't find input {}", entityDescriptor, e);
         }

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -20,6 +20,10 @@ import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.graph.Graph;
+import com.google.common.graph.GraphBuilder;
+import com.google.common.graph.ImmutableGraph;
+import com.google.common.graph.MutableGraph;
 import com.google.common.primitives.Ints;
 import org.graylog2.contentpacks.exceptions.ContentPackException;
 import org.graylog2.contentpacks.model.ModelId;
@@ -44,6 +48,10 @@ import org.graylog2.inputs.Input;
 import org.graylog2.inputs.InputService;
 import org.graylog2.inputs.converters.ConverterFactory;
 import org.graylog2.inputs.extractors.ExtractorFactory;
+import org.graylog2.inputs.extractors.LookupTableExtractor;
+import org.graylog2.lookup.LookupTable;
+import org.graylog2.lookup.db.DBLookupTableService;
+import org.graylog2.lookup.dto.LookupTableDto;
 import org.graylog2.plugin.Message;
 import org.graylog2.plugin.PluginMetaData;
 import org.graylog2.plugin.ServerStatus;
@@ -80,6 +88,7 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
 
     private final ObjectMapper objectMapper;
     private final InputService inputService;
+    private final DBLookupTableService lookupTableService;
     private final InputRegistry inputRegistry;
     private final MessageInputFactory messageInputFactory;
     private final ExtractorFactory extractorFactory;
@@ -92,6 +101,7 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
     public InputFacade(ObjectMapper objectMapper,
                        InputService inputService,
                        InputRegistry inputRegistry,
+                       DBLookupTableService lookupTableService,
                        MessageInputFactory messageInputFactory,
                        ExtractorFactory extractorFactory,
                        ConverterFactory converterFactory,
@@ -100,6 +110,7 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                        Map<String, MessageInput.Factory<? extends MessageInput>> inputFactories) {
         this.objectMapper = objectMapper;
         this.inputService = inputService;
+        this.lookupTableService = lookupTableService;
         this.inputRegistry = inputRegistry;
         this.messageInputFactory = messageInputFactory;
         this.extractorFactory = extractorFactory;
@@ -449,5 +460,32 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
         } catch (NotFoundException e) {
             return Optional.empty();
         }
+    }
+
+    @Override
+    public Graph<EntityDescriptor> resolveNativeEntity(EntityDescriptor entityDescriptor) {
+        final MutableGraph<EntityDescriptor> mutableGraph = GraphBuilder.directed().build();
+        mutableGraph.addNode(entityDescriptor);
+
+        final ModelId modelId = entityDescriptor.id();
+        try {
+            final Input input = inputService.find(modelId.toString());
+            final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));
+            inputWithExtractors.extractors().stream()
+                    .filter(e -> e.getType().equals(Extractor.Type.LOOKUP_TABLE))
+                    .map(e -> (String) e.getExtractorConfig().get(LookupTableExtractor.CONFIG_LUT_NAME))
+                    .map(name -> lookupTableService.get(name))
+                    .forEach(optional -> {
+                        optional.ifPresent(lookupTableDto -> {
+                            EntityDescriptor lookupTable = EntityDescriptor.create(
+                                    ModelId.of(lookupTableDto.id()), ModelTypes.LOOKUP_TABLE_V1);
+                            mutableGraph.putEdge(entityDescriptor, lookupTable);
+                        });
+                    });
+
+        } catch (NotFoundException e) {
+            LOG.debug("Couldn't find input {}", entityDescriptor, e);
+        }
+        return ImmutableGraph.copyOf(mutableGraph);
     }
 }

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -514,6 +514,13 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                 .map(e -> e.configuration())
                 .map(c -> ((ValueReference) c.get(LookupTableExtractor.CONFIG_LUT_NAME)).asString(parameters))
                 .collect(Collectors.toSet());
+
+        input.extractors().stream().filter(e -> e.type().asString(parameters).
+                equals(Extractor.Type.LOOKUP_TABLE.toString())).map(c -> c.converters()
+                .stream().map(con -> ((String)(((ValueReference)((ReferenceMap) con.configuration()
+                        .get("config")).get("lookup_table_name")).asString()))).collect(Collectors.toSet()))
+                .forEach(lookupTableNames::addAll);
+
         entities.entrySet().stream()
                 .filter(x -> x.getValue().type().equals(ModelTypes.LOOKUP_TABLE_V1))
                 .filter(x -> {

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -486,7 +486,6 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                     });
 
             inputWithExtractors.extractors().stream()
-                    .filter(e -> e.getType().equals(Extractor.Type.LOOKUP_TABLE))
                     .forEach(e -> e.getConverters().stream().filter(c -> c.getType().equals(Converter.Type.LOOKUP_TABLE))
                             .map(c -> (String) c.getConfig().get("lookup_table_name"))
                             .map(name -> lookupTableService.get(name))
@@ -528,8 +527,7 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                 .map(c -> ((ValueReference) c.get(LookupTableExtractor.CONFIG_LUT_NAME)).asString(parameters))
                 .collect(Collectors.toSet());
 
-        input.extractors().stream().filter(e -> e.type().asString(parameters).
-                equals(Extractor.Type.LOOKUP_TABLE.toString())).map(c -> c.converters()
+        input.extractors().stream().map(c -> c.converters()
                 .stream().map(con -> ((ValueReference) con.configuration().get("lookup_table_name")).asString(parameters))
                 .collect(Collectors.toSet()))
                 .forEach(lookupTableNames::addAll);

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -26,7 +26,6 @@ import com.google.common.graph.ImmutableGraph;
 import com.google.common.graph.MutableGraph;
 import com.google.common.primitives.Ints;
 import org.graylog2.contentpacks.exceptions.ContentPackException;
-import org.graylog2.contentpacks.jackson.ReferenceConverter;
 import org.graylog2.contentpacks.model.ModelId;
 import org.graylog2.contentpacks.model.ModelType;
 import org.graylog2.contentpacks.model.ModelTypes;
@@ -51,9 +50,7 @@ import org.graylog2.inputs.InputService;
 import org.graylog2.inputs.converters.ConverterFactory;
 import org.graylog2.inputs.extractors.ExtractorFactory;
 import org.graylog2.inputs.extractors.LookupTableExtractor;
-import org.graylog2.lookup.LookupTable;
 import org.graylog2.lookup.db.DBLookupTableService;
-import org.graylog2.lookup.dto.LookupTableDto;
 import org.graylog2.plugin.Message;
 import org.graylog2.plugin.PluginMetaData;
 import org.graylog2.plugin.ServerStatus;
@@ -78,6 +75,7 @@ import java.util.Optional;
 import java.util.Set;
 import java.util.UUID;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import static com.google.common.base.Strings.isNullOrEmpty;
 import static org.graylog2.contentpacks.model.entities.references.ReferenceMapUtils.toReferenceMap;
@@ -473,29 +471,23 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
         try {
             final Input input = inputService.find(modelId.toString());
             final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));
-            inputWithExtractors.extractors().stream()
+            final Stream<String> extractorLookupNames = inputWithExtractors.extractors().stream()
                     .filter(e -> e.getType().equals(Extractor.Type.LOOKUP_TABLE))
-                    .map(e -> (String) e.getExtractorConfig().get(LookupTableExtractor.CONFIG_LUT_NAME))
-                    .map(name -> lookupTableService.get(name))
-                    .forEach(optional -> {
-                        optional.ifPresent(lookupTableDto -> {
-                            EntityDescriptor lookupTable = EntityDescriptor.create(
-                                    ModelId.of(lookupTableDto.id()), ModelTypes.LOOKUP_TABLE_V1);
-                            mutableGraph.putEdge(entityDescriptor, lookupTable);
-                        });
-                    });
+                    .map(e -> (String) e.getExtractorConfig().get(LookupTableExtractor.CONFIG_LUT_NAME));
+            final Stream<String> converterLookupNames = inputWithExtractors.extractors().stream()
+                    .flatMap(e -> e.getConverters().stream())
+                    .filter(c -> c.getType().equals(Converter.Type.LOOKUP_TABLE))
+                    .map(c -> (String) c.getConfig().get("lookup_table_name"));
 
-            inputWithExtractors.extractors().stream()
-                    .forEach(e -> e.getConverters().stream().filter(c -> c.getType().equals(Converter.Type.LOOKUP_TABLE))
-                            .map(c -> (String) c.getConfig().get("lookup_table_name"))
-                            .map(name -> lookupTableService.get(name))
-                            .forEach(optional -> {
-                                optional.ifPresent(lookupTableDto -> {
-                                    EntityDescriptor lookupTable = EntityDescriptor.create(
-                                            ModelId.of(lookupTableDto.id()), ModelTypes.LOOKUP_TABLE_V1);
-                                    mutableGraph.putEdge(entityDescriptor, lookupTable);
-                                });
-                            }));
+            Stream.concat(extractorLookupNames, converterLookupNames)
+                    .map(lookupTableService::get)
+                    .filter(Optional::isPresent)
+                    .map(Optional::get)
+                    .forEach(lookupTableDto -> {
+                        EntityDescriptor lookupTable = EntityDescriptor.create(
+                                ModelId.of(lookupTableDto.id()), ModelTypes.LOOKUP_TABLE_V1);
+                        mutableGraph.putEdge(entityDescriptor, lookupTable);
+                    });
 
         } catch (NotFoundException e) {
             LOG.debug("Couldn't find input {}", entityDescriptor, e);
@@ -527,12 +519,10 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                 .map(c -> ((ValueReference) c.get(LookupTableExtractor.CONFIG_LUT_NAME)).asString(parameters))
                 .collect(Collectors.toSet());
 
-        input.extractors().stream().map(c -> c.converters()
-                .stream()
+        input.extractors().stream().flatMap(c -> c.converters().stream())
                 .filter(con -> con.type().asString(parameters).equals(Converter.Type.LOOKUP_TABLE.name()))
                 .map(con -> ((ValueReference) con.configuration().get("lookup_table_name")).asString(parameters))
-                .collect(Collectors.toSet()))
-                .forEach(lookupTableNames::addAll);
+                .forEach(lookupTableNames::add);
 
         entities.entrySet().stream()
                 .filter(x -> x.getValue().type().equals(ModelTypes.LOOKUP_TABLE_V1))

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -505,7 +505,7 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
     private Graph<Entity> resolveForInstallationV1(EntityV1 entity,
                                                    Map<String, ValueReference> parameters,
                                                    Map<EntityDescriptor, Entity> entities) {
-        final MutableGraph graph = GraphBuilder.directed().build();
+        final MutableGraph<Entity> graph = GraphBuilder.directed().build();
         graph.addNode(entity);
 
         final InputEntity input = objectMapper.convertValue(entity.data(), InputEntity.class);

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -530,8 +530,8 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
 
         input.extractors().stream().filter(e -> e.type().asString(parameters).
                 equals(Extractor.Type.LOOKUP_TABLE.toString())).map(c -> c.converters()
-                .stream().map(con -> ((String)(((ValueReference)((ReferenceMap) con.configuration()
-                        .get("config")).get("lookup_table_name")).asString()))).collect(Collectors.toSet()))
+                .stream().map(con -> ((ValueReference) con.configuration().get("lookup_table_name")).asString(parameters))
+                .collect(Collectors.toSet()))
                 .forEach(lookupTableNames::addAll);
 
         entities.entrySet().stream()

--- a/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ b/graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -515,7 +515,7 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
         final InputEntity input = objectMapper.convertValue(entity.data(), InputEntity.class);
         final Set<String> lookupTableNames = input.extractors().stream()
                 .filter(e -> e.type().asString(parameters).equals(Extractor.Type.LOOKUP_TABLE.toString()))
-                .map(e -> e.configuration())
+                .map(ExtractorEntity::configuration)
                 .map(c -> ((ValueReference) c.get(LookupTableExtractor.CONFIG_LUT_NAME)).asString(parameters))
                 .collect(Collectors.toSet());
 

--- a/graylog2-server/src/main/java/org/graylog2/inputs/extractors/LookupTableExtractor.java
+++ b/graylog2-server/src/main/java/org/graylog2/inputs/extractors/LookupTableExtractor.java
@@ -30,8 +30,8 @@ import java.util.Map;
 import static com.google.common.base.Strings.isNullOrEmpty;
 
 public class LookupTableExtractor extends Extractor {
-    private static final String CONFIG_LUT_NAME = "lookup_table_name";
     private final LookupTableService.Function lookupTable;
+    public static final String CONFIG_LUT_NAME = "lookup_table_name";
 
     public LookupTableExtractor(final MetricRegistry metricRegistry,
                                 final LookupTableService lookupTableService,

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -114,6 +114,8 @@ public class InputFacadeTest {
     private DBLookupTableService dbLookupTableService;
     @Mock
     private LookupTableDto lookupTableWhois;
+    @Mock
+    private LookupTableDto lookupTableTor;
 
     @Mock
     private MessageInputFactory messageInputFactory;
@@ -375,17 +377,25 @@ public class InputFacadeTest {
     @UsingDataSet(locations = "/org/graylog2/contentpacks/inputs.json", loadStrategy = LoadStrategyEnum.CLEAN_INSERT)
     public void resolveNativeEntity() throws NotFoundException {
         when(lookupuptableBuilder.lookupTable("whois")).thenReturn(lookupuptableBuilder);
+        when(lookupuptableBuilder.lookupTable("tor-exit-node-list")).thenReturn(lookupuptableBuilder);
         when(lookupuptableBuilder.build()).thenReturn(lookupTable);
         when(lookupTableService.newBuilder()).thenReturn(lookupuptableBuilder);
         when(lookupTableService.hasTable("whois")).thenReturn(true);
-        when(dbLookupTableService.get("whois")).thenReturn(Optional.of(lookupTableWhois));
+        when(lookupTableService.hasTable("tor-exit-node-list")).thenReturn(true);
+
         when(lookupTableWhois.id()).thenReturn("dead-beef");
+        when(dbLookupTableService.get("whois")).thenReturn(Optional.of(lookupTableWhois));
+
+        when(lookupTableTor.id()).thenReturn("dead-feed");
+        when(dbLookupTableService.get("tor-exit-node-list")).thenReturn(Optional.of(lookupTableTor));
 
         final Input input = inputService.find("5ae2eb0a3d27464477f0fd8b");
         EntityDescriptor entityDescriptor = EntityDescriptor.create(ModelId.of(input.getId()), ModelTypes.INPUT_V1);
-        EntityDescriptor expectedEntitiyDescriptor = EntityDescriptor.create(ModelId.of("dead-beef"), ModelTypes.LOOKUP_TABLE_V1);
+        EntityDescriptor expectedEntitiyDescriptorWhois = EntityDescriptor.create(ModelId.of("dead-beef"), ModelTypes.LOOKUP_TABLE_V1);
+        EntityDescriptor expectedEntitiyDescriptorTor = EntityDescriptor.create(ModelId.of("dead-feed"), ModelTypes.LOOKUP_TABLE_V1);
         Graph<EntityDescriptor> graph = facade.resolveNativeEntity(entityDescriptor);
-        assertThat(graph.nodes()).contains(expectedEntitiyDescriptor);
+        assertThat(graph.nodes()).contains(expectedEntitiyDescriptorWhois);
+        assertThat(graph.nodes()).contains(expectedEntitiyDescriptorTor);
     }
 
     @Test

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -408,12 +408,10 @@ public class InputFacadeTest {
         when(lookupTableService.hasTable("whois")).thenReturn(true);
         when(lookupTableService.hasTable("tor-exit-node-list")).thenReturn(true);
         final Input input = inputService.find("5ae2eb0a3d27464477f0fd8b");
-        final Map<String, String> lookupTableConfig = new HashedMap(1);
+        final Map<String, Object> lookupTableConfig = new HashedMap(1);
         lookupTableConfig.put("lookup_table_name", "tor-exit-node-list");
-        final Map<String, Object> config = new HashMap<>(1);
-        config.put("config", lookupTableConfig);
         final ConverterEntity converterEntity = ConverterEntity.create(
-                ValueReference.of("lookup_table"), ReferenceMapUtils.toReferenceMap(config));
+                ValueReference.of("lookup_table"), ReferenceMapUtils.toReferenceMap(lookupTableConfig));
         final List<ConverterEntity> converterEntities = new ArrayList<>(1);
         converterEntities.add(converterEntity);
         final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -61,6 +61,7 @@ import org.graylog2.lookup.db.DBLookupTableService;
 import org.graylog2.lookup.dto.LookupTableDto;
 import org.graylog2.plugin.PluginMetaData;
 import org.graylog2.plugin.ServerStatus;
+import org.graylog2.plugin.inputs.Converter;
 import org.graylog2.plugin.inputs.Extractor;
 import org.graylog2.plugin.inputs.MessageInput;
 import org.graylog2.shared.SuppressForbidden;
@@ -411,7 +412,7 @@ public class InputFacadeTest {
         final Map<String, Object> lookupTableConfig = new HashedMap(1);
         lookupTableConfig.put("lookup_table_name", "tor-exit-node-list");
         final ConverterEntity converterEntity = ConverterEntity.create(
-                ValueReference.of("lookup_table"), ReferenceMapUtils.toReferenceMap(lookupTableConfig));
+                ValueReference.of(Converter.Type.LOOKUP_TABLE.name()), ReferenceMapUtils.toReferenceMap(lookupTableConfig));
         final List<ConverterEntity> converterEntities = new ArrayList<>(1);
         converterEntities.add(converterEntity);
         final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -26,6 +26,8 @@ import com.google.common.graph.ImmutableGraph;
 import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
 import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
 import com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb;
+import com.mongodb.client.model.ValidationLevel;
+import org.elasticsearch.common.recycler.Recycler;
 import org.graylog2.contentpacks.model.ModelId;
 import org.graylog2.contentpacks.model.ModelTypes;
 import org.graylog2.contentpacks.model.entities.Entity;
@@ -33,8 +35,11 @@ import org.graylog2.contentpacks.model.entities.EntityDescriptor;
 import org.graylog2.contentpacks.model.entities.EntityExcerpt;
 import org.graylog2.contentpacks.model.entities.EntityV1;
 import org.graylog2.contentpacks.model.entities.EntityWithConstraints;
+import org.graylog2.contentpacks.model.entities.ExtractorEntity;
 import org.graylog2.contentpacks.model.entities.InputEntity;
+import org.graylog2.contentpacks.model.entities.LookupTableEntity;
 import org.graylog2.contentpacks.model.entities.NativeEntity;
+import org.graylog2.contentpacks.model.entities.references.ReferenceMap;
 import org.graylog2.contentpacks.model.entities.references.ReferenceMapUtils;
 import org.graylog2.contentpacks.model.entities.references.ValueReference;
 import org.graylog2.dashboards.DashboardImpl;
@@ -50,8 +55,12 @@ import org.graylog2.inputs.InputService;
 import org.graylog2.inputs.InputServiceImpl;
 import org.graylog2.inputs.converters.ConverterFactory;
 import org.graylog2.inputs.extractors.ExtractorFactory;
+import org.graylog2.inputs.extractors.LookupTableExtractor;
 import org.graylog2.inputs.random.FakeHttpMessageInput;
 import org.graylog2.inputs.raw.udp.RawUDPInput;
+import org.graylog2.lookup.LookupDefaultMultiValue;
+import org.graylog2.lookup.LookupDefaultSingleValue;
+import org.graylog2.lookup.LookupDefaultValue;
 import org.graylog2.lookup.LookupTableService;
 import org.graylog2.lookup.db.DBLookupTableService;
 import org.graylog2.lookup.dto.LookupTableDto;
@@ -72,9 +81,12 @@ import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnit;
 import org.mockito.junit.MockitoRule;
 
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
@@ -374,5 +386,66 @@ public class InputFacadeTest {
         EntityDescriptor expectedEntitiyDescriptor = EntityDescriptor.create(ModelId.of("dead-beef"), ModelTypes.LOOKUP_TABLE_V1);
         Graph graph = facade.resolveNativeEntity(entityDescriptor);
         assertThat(graph.nodes()).contains(expectedEntitiyDescriptor);
+    }
+
+    @Test
+    @UsingDataSet(locations = "/org/graylog2/contentpacks/inputs.json", loadStrategy = LoadStrategyEnum.CLEAN_INSERT)
+    public void resolveForInstallation() throws NotFoundException {
+        when(lookupuptableBuilder.lookupTable("whois")).thenReturn(lookupuptableBuilder);
+        when(lookupuptableBuilder.build()).thenReturn(lookupTable);
+        when(lookupTableService.newBuilder()).thenReturn(lookupuptableBuilder);
+        when(lookupTableService.hasTable("whois")).thenReturn(true);
+        when(dbLookupTableService.get("whois")).thenReturn(Optional.of(lookupTableDto));
+        when(lookupTableDto.id()).thenReturn("dead-beef");
+        final Input input = inputService.find("5ae2eb0a3d27464477f0fd8b");
+        final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));
+        final LookupTableExtractor extractor = (LookupTableExtractor) inputWithExtractors.extractors().iterator().next();
+        final ExtractorEntity extractorEntity = ExtractorEntity.create(
+                ValueReference.of(extractor.getTitle()),
+                ValueReference.of(extractor.getType()),
+                ValueReference.of(extractor.getCursorStrategy()),
+                ValueReference.of(extractor.getTargetField()),
+                ValueReference.of(extractor.getSourceField()),
+                ReferenceMapUtils.toReferenceMap(extractor.getExtractorConfig()),
+                Collections.emptyList(),
+                ValueReference.of(extractor.getConditionType()),
+                ValueReference.of(extractor.getConditionValue()),
+                ValueReference.of(extractor.getOrder())
+        );
+        List<ExtractorEntity> extractors = new ArrayList();
+        extractors.add(extractorEntity);
+        InputEntity inputEntity = InputEntity.create(
+                ValueReference.of(input.getTitle()),
+                ReferenceMapUtils.toReferenceMap(input.getConfiguration()),
+                Collections.emptyMap(),
+                ValueReference.of(input.getType()),
+                ValueReference.of(input.isGlobal()),
+                extractors);
+        final Entity entity = EntityV1.builder()
+                .id(ModelId.of(input.getId()))
+                .type(ModelTypes.INPUT_V1)
+                .data(objectMapper.convertValue(inputEntity, JsonNode.class))
+                .build();
+        final LookupTableEntity lookupTableEntity = LookupTableEntity.create(
+                ValueReference.of("whois"),
+                ValueReference.of("title"),
+                ValueReference.of("description"),
+                ValueReference.of("cache_name"),
+                ValueReference.of("dataadapter_name"),
+                ValueReference.of("default_single_value"),
+                ValueReference.of("BOOLEAN"),
+                ValueReference.of("default_multi_value"),
+                ValueReference.of("BOOLEAN")
+        );
+        final Entity expectedEntity = EntityV1.builder()
+                .id(ModelId.of("dead-beef"))
+                .data(objectMapper.convertValue(lookupTableEntity, JsonNode.class))
+                .type(ModelTypes.LOOKUP_TABLE_V1)
+                .build();
+        final EntityDescriptor entityDescriptor = expectedEntity.toEntityDescriptor();
+        final Map<EntityDescriptor, Entity> entityDescriptorEntityMap = new HashMap<>(1);
+        entityDescriptorEntityMap.put(entityDescriptor, expectedEntity);
+        Graph graph = facade.resolveForInstallation(entity, Collections.emptyMap(), entityDescriptorEntityMap);
+        assertThat(graph.nodes()).contains(expectedEntity);
     }
 }

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -275,8 +275,14 @@ public class InputFacadeTest {
                 .title("Local Raw UDP")
                 .build();
 
+        final EntityExcerpt expectedEntityExcerpt3 = EntityExcerpt.builder()
+                .id(ModelId.of("5ae2eb0a3d27464477f0fd8b"))
+                .type(ModelTypes.INPUT_V1)
+                .title("TEST PLAIN TEXT")
+                .build();
+
         final Set<EntityExcerpt> entityExcerpts = facade.listEntityExcerpts();
-        assertThat(entityExcerpts).containsOnly(expectedEntityExcerpt1, expectedEntityExcerpt2);
+        assertThat(entityExcerpts).containsOnly(expectedEntityExcerpt1, expectedEntityExcerpt2, expectedEntityExcerpt3);
     }
 
     @Test
@@ -363,10 +369,10 @@ public class InputFacadeTest {
         final Input input = inputService.find("5acc84f84b900a4ff290d9a7");
         final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input);
 
-        assertThat(inputService.totalCount()).isEqualTo(2L);
+        assertThat(inputService.totalCount()).isEqualTo(3L);
         facade.delete(inputWithExtractors);
 
-        assertThat(inputService.totalCount()).isEqualTo(1L);
+        assertThat(inputService.totalCount()).isEqualTo(2L);
         assertThatThrownBy(() -> inputService.find("5acc84f84b900a4ff290d9a7"))
                 .isInstanceOf(NotFoundException.class);
     }

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -22,6 +22,7 @@ import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.graph.Graph;
+import com.google.common.graph.ImmutableGraph;
 import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
 import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
 import com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb;
@@ -52,6 +53,8 @@ import org.graylog2.inputs.extractors.ExtractorFactory;
 import org.graylog2.inputs.random.FakeHttpMessageInput;
 import org.graylog2.inputs.raw.udp.RawUDPInput;
 import org.graylog2.lookup.LookupTableService;
+import org.graylog2.lookup.db.DBLookupTableService;
+import org.graylog2.lookup.dto.LookupTableDto;
 import org.graylog2.plugin.PluginMetaData;
 import org.graylog2.plugin.ServerStatus;
 import org.graylog2.plugin.inputs.Extractor;
@@ -98,6 +101,15 @@ public class InputFacadeTest {
     @Mock
     private LookupTableService lookupTableService;
     @Mock
+    private LookupTableService.Builder lookupuptableBuilder;
+    @Mock
+    private LookupTableService.Function lookupTable;
+    @Mock
+    private DBLookupTableService dbLookupTableService;
+    @Mock
+    private LookupTableDto lookupTableDto;
+
+    @Mock
     private MessageInputFactory messageInputFactory;
     @Mock
     private ServerStatus serverStatus;
@@ -133,6 +145,7 @@ public class InputFacadeTest {
                 objectMapper,
                 inputService,
                 inputRegistry,
+                dbLookupTableService,
                 messageInputFactory,
                 extractorFactory,
                 converterFactory,
@@ -346,4 +359,20 @@ public class InputFacadeTest {
                 .isInstanceOf(NotFoundException.class);
     }
 
+    @Test
+    @UsingDataSet(locations = "/org/graylog2/contentpacks/inputs.json", loadStrategy = LoadStrategyEnum.CLEAN_INSERT)
+    public void resolveNativeEntity() throws NotFoundException {
+        when(lookupuptableBuilder.lookupTable("whois")).thenReturn(lookupuptableBuilder);
+        when(lookupuptableBuilder.build()).thenReturn(lookupTable);
+        when(lookupTableService.newBuilder()).thenReturn(lookupuptableBuilder);
+        when(lookupTableService.hasTable("whois")).thenReturn(true);
+        when(dbLookupTableService.get("whois")).thenReturn(Optional.of(lookupTableDto));
+        when(lookupTableDto.id()).thenReturn("dead-beef");
+
+        final Input input = inputService.find("5ae2eb0a3d27464477f0fd8b");
+        EntityDescriptor entityDescriptor = EntityDescriptor.create(ModelId.of(input.getId()), ModelTypes.INPUT_V1);
+        EntityDescriptor expectedEntitiyDescriptor = EntityDescriptor.create(ModelId.of("dead-beef"), ModelTypes.LOOKUP_TABLE_V1);
+        Graph graph = facade.resolveNativeEntity(entityDescriptor);
+        assertThat(graph.nodes()).contains(expectedEntitiyDescriptor);
+    }
 }

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -22,14 +22,13 @@ import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.graph.Graph;
-import com.google.common.graph.ImmutableGraph;
 import com.lordofthejars.nosqlunit.annotation.UsingDataSet;
 import com.lordofthejars.nosqlunit.core.LoadStrategyEnum;
 import com.lordofthejars.nosqlunit.mongodb.InMemoryMongoDb;
-import com.mongodb.client.model.ValidationLevel;
-import org.elasticsearch.common.recycler.Recycler;
+import org.apache.commons.collections.map.HashedMap;
 import org.graylog2.contentpacks.model.ModelId;
 import org.graylog2.contentpacks.model.ModelTypes;
+import org.graylog2.contentpacks.model.entities.ConverterEntity;
 import org.graylog2.contentpacks.model.entities.Entity;
 import org.graylog2.contentpacks.model.entities.EntityDescriptor;
 import org.graylog2.contentpacks.model.entities.EntityExcerpt;
@@ -39,7 +38,6 @@ import org.graylog2.contentpacks.model.entities.ExtractorEntity;
 import org.graylog2.contentpacks.model.entities.InputEntity;
 import org.graylog2.contentpacks.model.entities.LookupTableEntity;
 import org.graylog2.contentpacks.model.entities.NativeEntity;
-import org.graylog2.contentpacks.model.entities.references.ReferenceMap;
 import org.graylog2.contentpacks.model.entities.references.ReferenceMapUtils;
 import org.graylog2.contentpacks.model.entities.references.ValueReference;
 import org.graylog2.dashboards.DashboardImpl;
@@ -58,9 +56,6 @@ import org.graylog2.inputs.extractors.ExtractorFactory;
 import org.graylog2.inputs.extractors.LookupTableExtractor;
 import org.graylog2.inputs.random.FakeHttpMessageInput;
 import org.graylog2.inputs.raw.udp.RawUDPInput;
-import org.graylog2.lookup.LookupDefaultMultiValue;
-import org.graylog2.lookup.LookupDefaultSingleValue;
-import org.graylog2.lookup.LookupDefaultValue;
 import org.graylog2.lookup.LookupTableService;
 import org.graylog2.lookup.db.DBLookupTableService;
 import org.graylog2.lookup.dto.LookupTableDto;
@@ -82,7 +77,6 @@ import org.mockito.junit.MockitoJUnit;
 import org.mockito.junit.MockitoRule;
 
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -119,7 +113,7 @@ public class InputFacadeTest {
     @Mock
     private DBLookupTableService dbLookupTableService;
     @Mock
-    private LookupTableDto lookupTableDto;
+    private LookupTableDto lookupTableWhois;
 
     @Mock
     private MessageInputFactory messageInputFactory;
@@ -384,8 +378,8 @@ public class InputFacadeTest {
         when(lookupuptableBuilder.build()).thenReturn(lookupTable);
         when(lookupTableService.newBuilder()).thenReturn(lookupuptableBuilder);
         when(lookupTableService.hasTable("whois")).thenReturn(true);
-        when(dbLookupTableService.get("whois")).thenReturn(Optional.of(lookupTableDto));
-        when(lookupTableDto.id()).thenReturn("dead-beef");
+        when(dbLookupTableService.get("whois")).thenReturn(Optional.of(lookupTableWhois));
+        when(lookupTableWhois.id()).thenReturn("dead-beef");
 
         final Input input = inputService.find("5ae2eb0a3d27464477f0fd8b");
         EntityDescriptor entityDescriptor = EntityDescriptor.create(ModelId.of(input.getId()), ModelTypes.INPUT_V1);
@@ -398,12 +392,20 @@ public class InputFacadeTest {
     @UsingDataSet(locations = "/org/graylog2/contentpacks/inputs.json", loadStrategy = LoadStrategyEnum.CLEAN_INSERT)
     public void resolveForInstallation() throws NotFoundException {
         when(lookupuptableBuilder.lookupTable("whois")).thenReturn(lookupuptableBuilder);
+        when(lookupuptableBuilder.lookupTable("tor-exit-node-list")).thenReturn(lookupuptableBuilder);
         when(lookupuptableBuilder.build()).thenReturn(lookupTable);
         when(lookupTableService.newBuilder()).thenReturn(lookupuptableBuilder);
         when(lookupTableService.hasTable("whois")).thenReturn(true);
-        when(dbLookupTableService.get("whois")).thenReturn(Optional.of(lookupTableDto));
-        when(lookupTableDto.id()).thenReturn("dead-beef");
+        when(lookupTableService.hasTable("tor-exit-node-list")).thenReturn(true);
         final Input input = inputService.find("5ae2eb0a3d27464477f0fd8b");
+        final Map<String, String> lookupTableConfig = new HashedMap(1);
+        lookupTableConfig.put("lookup_table_name", "tor-exit-node-list");
+        final Map<String, Object> config = new HashMap<>(1);
+        config.put("config", lookupTableConfig);
+        final ConverterEntity converterEntity = ConverterEntity.create(
+                ValueReference.of("lookup_table"), ReferenceMapUtils.toReferenceMap(config));
+        final List<ConverterEntity> converterEntities = new ArrayList<>(1);
+        converterEntities.add(converterEntity);
         final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));
         final LookupTableExtractor extractor = (LookupTableExtractor) inputWithExtractors.extractors().iterator().next();
         final ExtractorEntity extractorEntity = ExtractorEntity.create(
@@ -413,7 +415,7 @@ public class InputFacadeTest {
                 ValueReference.of(extractor.getTargetField()),
                 ValueReference.of(extractor.getSourceField()),
                 ReferenceMapUtils.toReferenceMap(extractor.getExtractorConfig()),
-                Collections.emptyList(),
+                converterEntities,
                 ValueReference.of(extractor.getConditionType()),
                 ValueReference.of(extractor.getConditionValue()),
                 ValueReference.of(extractor.getOrder())
@@ -432,7 +434,7 @@ public class InputFacadeTest {
                 .type(ModelTypes.INPUT_V1)
                 .data(objectMapper.convertValue(inputEntity, JsonNode.class))
                 .build();
-        final LookupTableEntity lookupTableEntity = LookupTableEntity.create(
+        final LookupTableEntity whoIsEntity = LookupTableEntity.create(
                 ValueReference.of("whois"),
                 ValueReference.of("title"),
                 ValueReference.of("description"),
@@ -443,15 +445,36 @@ public class InputFacadeTest {
                 ValueReference.of("default_multi_value"),
                 ValueReference.of("BOOLEAN")
         );
-        final Entity expectedEntity = EntityV1.builder()
+
+        final LookupTableEntity torNodeEntity = LookupTableEntity.create(
+                ValueReference.of("tor-exit-node-list"),
+                ValueReference.of("title"),
+                ValueReference.of("description"),
+                ValueReference.of("cache_name"),
+                ValueReference.of("dataadapter_name"),
+                ValueReference.of("default_single_value"),
+                ValueReference.of("BOOLEAN"),
+                ValueReference.of("default_multi_value"),
+                ValueReference.of("BOOLEAN")
+        );
+        final Entity expectedWhoIsEntity = EntityV1.builder()
                 .id(ModelId.of("dead-beef"))
-                .data(objectMapper.convertValue(lookupTableEntity, JsonNode.class))
+                .data(objectMapper.convertValue(whoIsEntity, JsonNode.class))
                 .type(ModelTypes.LOOKUP_TABLE_V1)
                 .build();
-        final EntityDescriptor entityDescriptor = expectedEntity.toEntityDescriptor();
-        final Map<EntityDescriptor, Entity> entityDescriptorEntityMap = new HashMap<>(1);
-        entityDescriptorEntityMap.put(entityDescriptor, expectedEntity);
+
+        final Entity expectedTorEntity = EntityV1.builder()
+                .id(ModelId.of("dead-feed"))
+                .data(objectMapper.convertValue(torNodeEntity, JsonNode.class))
+                .type(ModelTypes.LOOKUP_TABLE_V1)
+                .build();
+        final EntityDescriptor whoisDescriptor = expectedWhoIsEntity.toEntityDescriptor();
+        final EntityDescriptor torDescriptor = expectedTorEntity.toEntityDescriptor();
+        final Map<EntityDescriptor, Entity> entityDescriptorEntityMap = new HashMap<>(2);
+        entityDescriptorEntityMap.put(whoisDescriptor, expectedWhoIsEntity);
+        entityDescriptorEntityMap.put(torDescriptor, expectedTorEntity);
         Graph<Entity> graph = facade.resolveForInstallation(entity, Collections.emptyMap(), entityDescriptorEntityMap);
-        assertThat(graph.nodes()).contains(expectedEntity);
+        assertThat(graph.nodes()).contains(expectedWhoIsEntity);
+        assertThat(graph.nodes()).contains(expectedTorEntity);
     }
 }

--- a/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
+++ b/graylog2-server/src/test/java/org/graylog2/contentpacks/facades/InputFacadeTest.java
@@ -390,7 +390,7 @@ public class InputFacadeTest {
         final Input input = inputService.find("5ae2eb0a3d27464477f0fd8b");
         EntityDescriptor entityDescriptor = EntityDescriptor.create(ModelId.of(input.getId()), ModelTypes.INPUT_V1);
         EntityDescriptor expectedEntitiyDescriptor = EntityDescriptor.create(ModelId.of("dead-beef"), ModelTypes.LOOKUP_TABLE_V1);
-        Graph graph = facade.resolveNativeEntity(entityDescriptor);
+        Graph<EntityDescriptor> graph = facade.resolveNativeEntity(entityDescriptor);
         assertThat(graph.nodes()).contains(expectedEntitiyDescriptor);
     }
 
@@ -418,7 +418,7 @@ public class InputFacadeTest {
                 ValueReference.of(extractor.getConditionValue()),
                 ValueReference.of(extractor.getOrder())
         );
-        List<ExtractorEntity> extractors = new ArrayList();
+        List<ExtractorEntity> extractors = new ArrayList<>();
         extractors.add(extractorEntity);
         InputEntity inputEntity = InputEntity.create(
                 ValueReference.of(input.getTitle()),
@@ -451,7 +451,7 @@ public class InputFacadeTest {
         final EntityDescriptor entityDescriptor = expectedEntity.toEntityDescriptor();
         final Map<EntityDescriptor, Entity> entityDescriptorEntityMap = new HashMap<>(1);
         entityDescriptorEntityMap.put(entityDescriptor, expectedEntity);
-        Graph graph = facade.resolveForInstallation(entity, Collections.emptyMap(), entityDescriptorEntityMap);
+        Graph<Entity> graph = facade.resolveForInstallation(entity, Collections.emptyMap(), entityDescriptorEntityMap);
         assertThat(graph.nodes()).contains(expectedEntity);
     }
 }

--- a/graylog2-server/src/test/resources/org/graylog2/contentpacks/inputs.json
+++ b/graylog2-server/src/test/resources/org/graylog2/contentpacks/inputs.json
@@ -150,6 +150,55 @@
           "key": "custom_field"
         }
       ]
+    },
+    {
+      "_id": {
+        "$oid": "5ae2eb0a3d27464477f0fd8b"
+      },
+      "creator_user_id" : "admin",
+      "configuration" : {
+        "recv_buffer_size" : 1048576,
+        "tcp_keepalive" : false,
+        "use_null_delimiter" : false,
+        "number_worker_threads" : 8,
+        "tls_client_auth_cert_file" : "",
+        "bind_address" : "127.0.0.1",
+        "tls_cert_file" : "",
+        "port" : 5555,
+        "tls_key_file" : "",
+        "tls_enable" : false,
+        "tls_key_password" : "",
+        "max_message_size" : 2097152,
+        "tls_client_auth" : "disabled",
+        "override_source" : null
+      },
+      "name" : "Raw/Plaintext TCP",
+      "created_at": {
+        "$date": "2018-10-18T12:38:01.645Z"
+      },
+      "global" : false,
+      "type" : "org.graylog2.inputs.raw.tcp.RawTCPInput",
+      "title" : "TEST PLAIN TEXT",
+      "content_pack" : null,
+      "node_id" : "f3bd37d4-61d2-4157-8d72-8b26d1777c3f",
+      "extractors" : [
+        {
+          "creator_user_id" : "admin",
+          "source_field" : "request_ip",
+          "condition_type" : "none",
+          "title" : "WHOIS-Extractor",
+          "type" : "lookup_table",
+          "cursor_strategy" : "copy",
+          "target_field" : "WHOIS",
+          "extractor_config" : {
+            "lookup_table_name" : "whois"
+          },
+          "condition_value" : "",
+          "converters" : [],
+          "id" : "7e7aee30-d212-11e8-bda9-00e18cb9c35a",
+        }
+      ]
     }
+
   ]
 }

--- a/graylog2-server/src/test/resources/org/graylog2/contentpacks/inputs.json
+++ b/graylog2-server/src/test/resources/org/graylog2/contentpacks/inputs.json
@@ -194,8 +194,15 @@
             "lookup_table_name" : "whois"
           },
           "condition_value" : "",
-          "converters" : [],
-          "id" : "7e7aee30-d212-11e8-bda9-00e18cb9c35a",
+          "converters" : [
+            {
+              "type" : "lookup_table",
+              "config" : {
+                "lookup_table_name" : "tor-exit-node-list"
+              }
+            }
+          ],
+          "id" : "7e7aee30-d212-11e8-bda9-00e18cb9c35a"
         }
       ]
     }

